server/src/main/java/com/focusit/service/JMeterRecorderService.java

```diff
@@ -50,24 +49,21 @@ public void startJMeter(MongoDbScenario scenario) throws Exception
         {
             return;
         }
-        if (jmeterStartStopLock.tryLock() || jmeterStartStopLock.tryLock(10, TimeUnit.SECONDS))
+        if (!jmeterStartStopLock.tryLock() && !jmeterStartStopLock.tryLock(10, TimeUnit.SECONDS))
         {
-            if (availablePorts.size() < 0)
-            {
-                throw new IllegalArgumentException("No ports left to start JMeter");
-            }
+            LOG.error("Can't acquire a lock to start JMeter");
+            throw new IllegalStateException("Can't acquire a lock to start JMeter");
         }
-        else
+        if (availablePorts.isEmpty())
         {
-            LOG.error("Can't acquire a lock to start JMeter");
+            throw new IllegalStateException("No ports left to start JMeter");
         }
         try
         {
-            int port = availablePorts.get(0);
-            availablePorts.remove(0);
+            int port = availablePorts.remove(0);
             scenario.getConfiguration().getCommonConfiguration().setProxyPort("" + port);
-            JMeterRecorder recorder = new JMeterRecorder(scenario.getConfiguration().getCommonConfiguration()
-                    .getScriptClassloader());
+            ScriptEngine.init(scenario.getConfiguration().getCommonConfiguration().getScriptClassloader());
```

Comment:
Sometimes it is needed to start playing without recording JMeter scenario(empty port in configuration, though it is mostly for debugging recording), in this case ScriptEngine will remain uninitialized and cause problems. 
ScriptEngine should be initialized elsewhere